impexp-kml-collada-plugin/src/main/java/org/citydb/modules/kml/database/KmlExporterManager.java

```diff
@@ -650,23 +663,137 @@ public void print(ColladaBundle colladaBundle, long id, boolean balloonInSeparat
+			commands.add("-o");
+			commands.add(gltfModelFile.getAbsolutePath());
+			commands.add("-v");
+			commands.add(exportGltfV1 ? "1.0" : "2.0");
+			if (!config.getProject().getKmlExporter().isEmbedTexturesInGltfFiles()) {
+				// always embed textures if glb is enabled, this is handled in the GUI
+				commands.add("-t");
+			}
+			if (config.getProject().getKmlExporter().isExportGltfBinary()) {
+				commands.add("-b");
+			}
+			if (config.getProject().getKmlExporter().isEnableGltfDracoCompression()) {
+				commands.add("-d");
+			}
+			ProcessBuilder pb = new ProcessBuilder(commands);
 			pb.directory(buildingDirectory);
 			try {
 				Process process = pb.start();
 				process.waitFor();
+				// change the image URI in the glTF file
+				if (!config.getProject().getKmlExporter().isExportGltfBinary() && !config.getProject().getKmlExporter().isEmbedTexturesInGltfFiles()) {
+					BufferedReader br = new BufferedReader(new FileReader(gltfModelFile.getAbsoluteFile()));
+					StringBuilder sb = new StringBuilder();
+					String line = br.readLine();
+
+					String texImageUri = "";
+
+					while (line != null) {
+						if (line.contains("images")) {
+							while (!line.contains("uri")) {
+								sb.append(line);
+								sb.append(System.lineSeparator());
+								line = br.readLine();
+							}
+							Pattern p = Pattern.compile("\"([^\"]*)\"");
+							Matcher m = p.matcher(line);
+							while (m.find()) {
+								String found = m.group(1);
+								if (!found.equals("uri")) {
+									texImageUri = m.group(1);
+									if (implicitId != null) {
+										line = line.replace(texImageUri, "../../../" + "Appearances" + "/" + texImageUri);
+									} else {
+										line = line.replace(texImageUri, "../../../../" + "Appearances" + "/" + texImageUri);
+									}
+								}
+							}
+						}
+						sb.append(line);
+						sb.append(System.lineSeparator());
+						line = br.readLine();
+					}
+					br.close();
+
+					Writer writer = new BufferedWriter(new FileWriter(gltfModelFile.getAbsolutePath()));
+					writer.write(sb.toString());
+					writer.close();
+
+					// remove the image file
+					(new File(buildingDirectory + File.separator + texImageUri)).delete();
+				}
 			} catch (IOException | InterruptedException e) {
-				log.debug("Unexpected errors occurred while converting collada to glTF for city object '" + colladaBundle.getGmlId() + "' with output path: '" + gltfModelFile.getAbsolutePath() + "'");
+				log.debug("Unexpected errors occurred while converting collada to glTF for city object '" + colladaBundle.getGmlId() + "' with output path: '" + gltfModelFile.getAbsolutePath() + "'" + "\n" + e.getMessage());
 			} finally {
-				if (config.getProject().getKmlExporter().isNotCreateColladaFiles() && gltfModelFile.exists()) {
+				if (config.getProject().getKmlExporter().isNotCreateColladaFiles() && (gltfModelFile.exists() || (new File(gltfModelFile.getAbsolutePath().replace(".gltf", ".glb"))).exists())) {
```

Comment:
Moved to branch [glb-draco-support](https://github.com/3dcitydb/importer-exporter/tree/glb-draco-support).